Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/LeaderBoard.hs

```diff
@@ -196,6 +194,61 @@ getWeeklyDriverLeaderBoard (personId, merchantId, merchantOpCityId) fromDate toD
       return $ LeaderBoardRes (currDriverInfo : drivers') (Just now) (fromIntegral totalEligibleDrivers)
     else return $ LeaderBoardRes drivers' (Just now) (fromIntegral totalEligibleDrivers)
 
+getMonthlyDriverLeaderBoard :: (Esq.EsqDBFlow m r, Esq.EsqDBReplicaFlow m r, EncFlow m r, Redis.HedisFlow m r, CacheFlow m r) => (Id Person, Id DM.Merchant, Id DMOC.MerchantOperatingCity) -> Day -> Day -> m LeaderBoardRes
+getMonthlyDriverLeaderBoard (personId, merchantId, merchantOpCityId) fromDate toDate = do
+  now <- getCurrentTime
+  let currentDay = RideEndInt.getCurrentDate now
+      monthDiff = getMonth currentDay - getMonth toDate
+      inputMonthDiff = getMonth toDate - getMonth fromDate
+  monthlyLeaderBoardConfig <- QLeaderConfig.findLeaderBoardConfigbyType LConfig.MONTHLY merchantOpCityId >>= fromMaybeM (InternalError "Leaderboard configs not present")
+  unless monthlyLeaderBoardConfig.isEnabled . throwError $ InvalidRequest "Leaderboard Not Available"
+  when (monthDiff < 0 || monthDiff > 3) $ throwError $ InvalidRequest "Month outside Range"
+  when (inputMonthDiff /= 0 || fromDate /= RideEndInt.getStartDateMonth fromDate || toDate /= RideEndInt.getEndDateMonth toDate 1) $
```

Comment:
Hmm. so no use of taking `fromDate` and `toDate` here as an input then? We could've just taken month number(1-12). Anyway, this is fine for now